Aplib.Core/Desire/GoalStructures/SequentialGoalStructure.cs

```diff
@@ -24,20 +24,25 @@ public class SequentialGoalStructure<TBeliefSet> : GoalStructure<TBeliefSet>, ID
         /// <summary>
         /// Initializes a new instance of the <see cref="SequentialGoalStructure{TBeliefSet}" /> class.
         /// </summary>
+        /// <param name="metadata">
+        /// Metadata about this GoalStructure, used to quickly display the goal in several contexts.
+        /// </param>
         /// <param name="children">The children of the goal structure.</param>
-        public SequentialGoalStructure(params IGoalStructure<TBeliefSet>[] children)
-            : base(children)
+        public SequentialGoalStructure(Metadata metadata, params IGoalStructure<TBeliefSet>[] children)
+            : base(metadata, children)
         {
             if (children.Length <= 0)
-            {
                 throw new ArgumentException("Collection of children is empty", nameof(children));
-            }
-
             _childrenEnumerator = _children.GetEnumerator();
```

Comment:
Nit: keep the newline?
```suggestion
            
            _childrenEnumerator = _children.GetEnumerator();
```